scIB/metrics/trajectory.py

```diff
@@ -60,44 +61,51 @@ def get_root(
         min_dpt_cell[opt(diffmap_min_dpt)] += 1
 
     # root cell is cell with max vote
-    return min_dpt[np.argmax(min_dpt_cell)], adata_post_sub
+    return min_dpt[np.argmax(min_dpt_cell)], adata_post_ti
 
 
 def trajectory_conservation(
         adata_pre,
         adata_post,
         label_key,
-        pseudotime_key="dpt_pseudotime"
+        pseudotime_key="dpt_pseudotime",
+	batch_key=None
 ):
     """
     :param adata_pre: unintegrated adata
     :param adata_post: integrated adata
     :param label_key: column in `adata_pre.obs` of the groups used to precompute the trajectory
     :param pseudotime_key: column in `adata_pre.obs` in which the pseudotime is saved in.
         Column can contain empty entries, the dataset will be subset to the cells with scores.
+    :param batch_key: set to batch key if if you want to compute the trajectory metric by batch
     """
     # subset to cells for which pseudotime has been computed
     cell_subset = adata_pre.obs.index[adata_pre.obs[pseudotime_key].notnull()]
-    adata_pre_sub = adata_pre[cell_subset]
-    adata_post_sub = adata_post[cell_subset]
-
+    adata_pre_ti = adata_pre[cell_subset]
+    adata_post_ti = adata_post[cell_subset]
     try:
-        iroot, adata_post_sub2 = get_root(adata_pre_sub, adata_post_sub, label_key, pseudotime_key)
+        iroot, adata_post_ti2 = get_root(adata_pre_ti, adata_post_ti, label_key, pseudotime_key)
     except RootCellError:
         print('No root cell found, setting trajectory conservation metric to 0.')
         return 0  # failure to find root cell means no TI conservation
 
-    adata_post_sub2.uns['iroot'] = iroot
+    adata_post_ti2.uns['iroot'] = iroot
 
-    sc.tl.dpt(adata_post_sub2)  # stored in 'dpt_pseudotime'
-    adata_post_sub2.obs.loc[adata_post_sub2.obs['dpt_pseudotime'] > 1, 'dpt_pseudotime'] = 0
-    adata_post_sub.obs['dpt_pseudotime'] = 0
-    adata_post_sub.obs['dpt_pseudotime'] = adata_post_sub2.obs['dpt_pseudotime']
-    adata_post_sub.obs['dpt_pseudotime'].fillna(0, inplace=True)
+    sc.tl.dpt(adata_post_ti2)  # stored in 'dpt_pseudotime'
+    adata_post_ti2.obs.loc[adata_post_ti2.obs['dpt_pseudotime'] > 1, 'dpt_pseudotime'] = 0
+    adata_post_ti.obs['dpt_pseudotime'] = 0
+    adata_post_ti.obs['dpt_pseudotime'] = adata_post_ti2.obs['dpt_pseudotime']
+    adata_post_ti.obs['dpt_pseudotime'].fillna(0, inplace=True)
 
-    pseudotime_before = adata_post_sub.obs['dpt_pseudotime']
-    pseudotime_after = adata_pre_sub.obs[pseudotime_key]
+    pseudotime_before = adata_post_ti.obs['dpt_pseudotime']
+    pseudotime_after = adata_pre_ti.obs[pseudotime_key]
     correlation = pseudotime_before.corr(pseudotime_after, 'spearman')
-
-    return (correlation + 1) / 2  # scaled
-
+    if batch_key == None:
+        return (correlation + 1) / 2  # scaled
+    else:
+        corr = pd.Series()
+        for i in batch_key:
```

Comment:
According to the documentation `batch_key` is a string that is the name of an `adata.obs` column. Here, it looks like it is `adata.obs[batch_key].unique()`. This needs to be adapted.

Reply:
yeah, should have actually tested the code